Source/JavaScriptCore/assembler/MacroAssemblerX86_64.h

```diff
@@ -3299,15 +3299,77 @@ class MacroAssemblerX86_64 : public MacroAssemblerX86Common {
         }
     }
 
-    void vectorLoad8Splat(Address address, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(dest); }
-    void vectorLoad16Splat(Address address, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(dest); }
-    void vectorLoad32Splat(Address address, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(dest); }
-    void vectorLoad64Splat(Address address, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(dest); }
-
-    void vectorLoad8Lane(Address address, TrustedImm32 imm, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(imm); UNUSED_PARAM(dest); }
-    void vectorLoad16Lane(Address address, TrustedImm32 imm, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(imm); UNUSED_PARAM(dest); }
-    void vectorLoad32Lane(Address address, TrustedImm32 imm, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(imm); UNUSED_PARAM(dest); }
-    void vectorLoad64Lane(Address address, TrustedImm32 imm, FPRegisterID dest) { UNUSED_PARAM(address); UNUSED_PARAM(imm); UNUSED_PARAM(dest); }
+    void vectorLoad8Splat(Address address, FPRegisterID dest, RegisterID scratch)
+    {
+        load8(address, scratch);
+        vectorSplat8(scratch, dest);
+    }
+
+    void vectorLoad16Splat(Address address, FPRegisterID dest, RegisterID scratch)
+    {
+        load16(address, scratch);
+        vectorSplat16(scratch, dest);
+    }
+
+    void vectorLoad32Splat(Address address, FPRegisterID dest, RegisterID scratch)
+    {
+        load32(address, scratch);
+        vectorSplat32(scratch, dest);
+    }
+
+    void vectorLoad64Splat(Address address, FPRegisterID dest, RegisterID scratch)
+    {
+        load64(address, scratch);
+        vectorSplat64(scratch, dest);
+    }
```

Comment:
I think we can implement them more efficiently. For example, how about using vmmoddup_mr?